test/document_stores/test_pinecone.py

```diff
@@ -402,3 +402,18 @@ def test_get_all_documents_extended_filter_compound_same_level_not(
 
         with pytest.raises(FilterError, match="Comparison value for '\$[l|g]te' operation must be a float or int."):
             doc_store_with_docs.get_all_documents(filters=filters)
+
+    @pytest.mark.integration
+    def test_multilayer_dict(self, doc_store_with_docs: PineconeDocumentStore):
+        # TODO add test that multilayer dict can be upserted
```

Comment:
Do you plan to address this TODO in this PR?

Reply:
That function does do this, I just forgot to remove "TODO" :)

Reply:
removed in next commit